plugins/destination/s3/client/write_test.go

```diff
@@ -44,35 +44,35 @@ func TestSanitizeJSONKeys(t *testing.T) {
 		},
 		"foo:bar":     "baz",
 		"foo-bar-baz": []any{"baz", map[string]any{"foo:bar": "baz"}},
-		"string": map[string]string{
+		"string": map[string]any{
 			"foo-bar": "baz",
 		},
-		"int": map[string]int{
+		"int": map[string]any{
 			"foo-bar": 123,
 		},
-		"pointer": map[string]*string{
-			"foo-bar": &[]string{"baz"}[0],
+		"pointer": map[string]any{
```

Comment:
we could instead go for `marshal -> unmarshal -> call sanitizeJSONKeysForObject` route